ocaml/typing/typecore.ml

```diff
@@ -7439,8 +7549,8 @@ and type_label_exp create env (arg_mode : expected_mode) loc ty_expected
   in
   (lid, label, arg)
 
-and type_argument ?explanation ?recarg env (mode : expected_mode) sarg
-      ty_expected' ty_expected =
+and type_argument ?explanation ?recarg ?(overwrite = No_overwrite) env
```

Comment:
But here this would touch even more places, most of which can never overwrite. Note that it is always sound to forget `overwrite`; it just means that the user can not write `_` in places where they might expect it. Furthermore, that would be an easy bug to find and fix.

Reply:
I guess my argument is that we don't just want to be sound: we want to implement the design faithfully, and that seems more likely to happen without optional arguments.